robotouille/robotouille_simulator.py

```diff
@@ -1,33 +1,51 @@
 import pygame
+import time
 from utils.robotouille_input import create_action_from_control
 from robotouille.robotouille_env import create_robotouille_env
+from backend.movement.player import Player
+from backend.movement.movement import Movement, Mode
 
 
 def simulator(environment_name: str, seed: int=42, noisy_randomization: bool=False):
     # Your code for robotouille goes here
-    env, json, renderer = create_robotouille_env(environment_name, seed, noisy_randomization)
+    movement_mode = Mode.TRAVERSE
+    env, json, renderer = create_robotouille_env(environment_name, movement_mode, seed, noisy_randomization)
     obs, info = env.reset()
     renderer.render(obs, mode='human')
     done = False
     interactive = False # Set to True to interact with the environment through terminal REPL (ignores input)
     
+    players = obs.get_players()
+    actions = []
     while not done:
-        # Construct action from input
         pygame_events = pygame.event.get()
-        # Mouse clicks for movement and pick/place stack/unstack
         mousedown_events = list(filter(lambda e: e.type == pygame.MOUSEBUTTONDOWN, pygame_events))
-        # Keyboard events ('e' button) for cut/cook ('space' button) for noop
         keydown_events = list(filter(lambda e: e.type == pygame.KEYDOWN, pygame_events))
-        actions = []
-        action, args = create_action_from_control(env, obs, obs.current_player, mousedown_events+keydown_events, renderer)
-        for player in obs.get_players():
-            if player == obs.current_player:
-                actions.append((action, args))
-            else:
-                actions.append((None, None))
-        if not interactive and action is None:
+        player_obj = Player.get_player(obs.current_player.name)
+        no_action = True
+        
+        # If player is moving, do not allow any action; action will be None
+        if Movement.is_player_moving(player_obj.name):
```

Comment:
no need for a `no_action` boolean, try this
```
# If player is moving, skip actions
if Movement.is_player_moving(player_obj.name):
  continue
action, args = create_action_from_control(env, obs, obs.current_player, mousedown_events+keydown_events, renderer)
if not interactive and action is None:
     # Retry for keyboard input
     continue
```
I believe you are trying to take environment steps because of the `clock.tick(20)` calls in movement but we fixed this by adding `env.render(...)` in this game loop which would just require the rendering to occur but the actual game timesteps to be fixed (assuming movement is always still 1 timestep)

Reply:
Issue made

Reply:
This code didnt work because currently we need to pass in a None action if a player is moving. This none action needs to be handled differently depending if the player is moving. If they are moving, we do not wait for another keyboard input and we step the environment. Else if they are moving and action is None, we retry for keyboard input.